src/exact/ExactMapper.cpp

```diff
@@ -327,13 +327,28 @@ void ExactMapper::map(const Configuration& settings) {
 
 void ExactMapper::coreMappingRoutine(const std::set<unsigned short>& qubitChoice, const CouplingMap& rcm, MappingResults& choiceResults, std::vector<std::vector<std::pair<unsigned short, unsigned short>>>& swaps, long unsigned int limit, unsigned int timeout) {
     const auto& config = results.config;
-
-    // Z3 context
-    context c;
+    using namespace logicbase;
+    // LogicBlock
+    LogicBlock* lb;
```

Comment:
```suggestion
    std::unique_ptr<LogicBlock> lb;
```
Use C++ facilities to automatically manage memory. This requires some changes in the LogicBlocks repository (specifically the `Encodings` files).
Instead of passing `LogicBlock*` either pass `std::unique_ptr<LogicBlock>&` or `LogicBlock&` and adapt the call sites in QMAP correspondingly.